File: c++/web_tutorial_mastercopy/elevation.cpp
#include "Bridges.h"
#include "DataSource.h"
#include <iostream>
#include <string>
#include "data_src/ElevationData.h"

using namespace std;
using namespace bridges;

int main(int argc, char **argv) {

	// create Bridges object
#if TESTING
	// command line args provide credentials and server to test on
	Bridges bridges (atoi(argv[1]), argv[2], argv[3]);

	if (argc > 4)
		bridges.setServer(argv[4]);

	bridges.setTitle("An Elevation Map Dataset Example");
#else
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
#endif

	//create the Bridges object, set credentials

	bridges.setServer("clone");

	DataSource ds;
	dataset::ElevationData *elev_data;
	try {
		//osm_data = ds->getOSMData(35.28, -80.75, 35.32, -80.71);
		elev_data = ds.getElevationData(6.020558108041891,46.10757941505641,
						9.707863253414155,47.77059916141684, 0.02);
	}
	catch (std::string s) {
		std::cerr << "Exception: " << s << "\n";
	}

	return 0;
}

File: python/web_tutorial_mastercopy/elevation.py
from bridges.color_grid import *
from bridges.bridges import *
from bridges.color import *
from bridges.data_src_dependent import data_source
import sys
import os

def main():

#if TESTING
    # command line args provide credentials and server to test on
    args = sys.argv[1:]
    bridges = Bridges(int(args[0]), args[1], args[2])
    if len(args) > 3:
        bridges.connector.set_server(args[3])
#else
    bridges = Bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY")
#endif


    ele_obj = data_source.get_elevation_data([-98.02593749997456, 41.03133177632377, -96.94531249997696, 42.508577297430456])
    print("width: "+str(ele_obj.cols))
    print("height: "+str(ele_obj.rows))
    print(ele_obj._xll)
    print(ele_obj._yll)
    print("spatial resolution: " + ele_obj.cellsize)

    print("first 10 elevations:")
    for i in range(0, 10):
        print(ele_obj.data[0][i])
    

    

if __name__ == "__main__":
    main()
